src/adapters/stargate/products/pool/stargatePoolAdapter.ts

```diff
@@ -84,78 +86,68 @@ export class StargatePoolAdapter implements IProtocolAdapter {
       tokens: this.protocolTokens,
     })
 
-    const underlying = await Promise.all(
+    const tokens = await Promise.all(
       protocolTokensBalances.map(async (protocolTokenBalance) => {
         const amountLPtoLD = await this.stargateTokenContract(
           protocolTokenBalance.address,
         ).amountLPtoLD(protocolTokenBalance.balanceRaw)
 
         const underlyingTokenMetadata =
-          this.metadata[protocolTokenBalance.address].underlying
+          this.metadata[protocolTokenBalance.address]!.underlying
```

Comment:
I can't see how to avoid the bang here without creating a lot of messy code. We know `getBalances` returns a balance for each lp in metadata, and we know that every metadata pool has one and only one underlying token.

Reply:
So getBalance also filters out 0 balances, which not an issue, as long as getBalance doesn't do anything funky like changing the lp address from '0x0000' to '0xpotatoe' then we should be ok

Reply:
Yes, as long as `getBalances` doesn't return anything that wasn't already on the list, we are good. It's a `utils` function, so it should not be normally changed by external contributors.